main/guides/zoe/contract-state.md

```diff
@@ -0,0 +1,35 @@
+# State Smart Contract
+
+In our first `greetings` smart contract, we created a `greet` function and exposed it using `publicFacet` so that it can be remotely called. However, if you notice, there is no state in our smart contract that is preserved between calls. Contracts can use ordinary variables and data structures for state. 
+
+In our second example smart contract, we will manage a list of rooms. We want everyone with access to `publicFacet` to be able to create a new room, and also get current count of rooms. We maintain state using `Map` data structure as below:
+
+<<< @/../snippets/zoe/src/02-state.js#rooms-map
+
+Anyone can add new rooms by making a call to `makeRoom` which is defined as:
+
+<<< @/../snippets/zoe/src/02-state.js#makeRoom
+
+Using `makeRoom` changes the results of the following call to `getRoomCount` which returns the number of rooms:
```

Comment:
This might help here:

```
const makeRoom = id => {
    let count = 0;
    const room = Far('Room', {
      getId: () => id,
      incr: () => (count += 1),
      decr: () => (count -= 1),
    });
    rooms.set(id, room);
    return room;
  };
```
Maybe change:
"Using `makeRoom` changes the results of the following call to `getRoomCount` which returns the number of rooms:

to

Using `makeRoom` creates a new room, exposing these functions to be invoked on the newly added room, ```getId```, ```incr```, and ```decr```. As you can see this pattern follows the Object Capability model, as whoever receives the room from invoking ```makeRoom```, will now have access to these three methods.

Then can finish here by stating once the room is complete, ```rooms.set(id, room);``` inputs the newly created room, into the contract's map state variable.

Overall, i'd like to see more inline comments in out example contracts, linking to external sources. Good job @mudassir-agoric

Reply:
Fixed!